<launch>
  <!-- Arguments -->
  <arg name="model" default="burger" doc="model type [burger, waffle, waffle_pi]"/>
  <arg name="cmd_vel_topic" default="/cmd_vel" />
  <arg name="odom_topic" default="odom" />
  <arg name="move_forward_only" default="true"/>

  <!-- move_base -->
  <node pkg="move_base" type="move_base" respawn="false" name="move_base" output="screen">
    <param name="base_local_planner" value="dwa_local_planner/DWAPlannerROS" />
    <!-- <param name="base_global_planner" value="startup/GlobalPlanner"/> -->
    <rosparam file="$(find controls_stack)/params/costmap_common_params_$(arg model).yaml" command="load" ns="global_costmap" />
    <rosparam file="$(find controls_stack)/params/costmap_common_params_$(arg model).yaml" command="load" ns="local_costmap" />
    <rosparam file="$(find controls_stack)/params/local_costmap_params.yaml" command="load" />
    <rosparam file="$(find controls_stack)/params/global_costmap_params.yaml" command="load" />
    <rosparam file="$(find controls_stack)/params/move_base_params.yaml" command="load" />
    <rosparam file="$(find controls_stack)/params/dwa_local_planner_params_burger.yaml" command="load" />
    <remap from="cmd_vel" to="$(arg cmd_vel_topic)"/>
    <remap from="odom" to="$(arg odom_topic)"/>
    <param name="DWAPlannerROS/min_vel_x" value="0.0" if="$(arg move_forward_only)" />
  </node>
</launch>
